include/modem/location.h

```diff
@@ -485,6 +485,34 @@ struct location_config {
 	 * location_config_defaults_set() function is called.
 	 */
 	enum location_req_mode mode;
+
```

Comment:
Our intention has been to keep the Location library API as simple as possible. Probably we don't want to have these options in the API at all. In any case, the current approach has several problems.

At least for now, I think the used configuration should be just hard coded in `cloud_service_nrf.c` like the `request_loc` parameter was earlier and all other Location library changes should be removed. Or what do you think, @trantanen?

Reply:
Yep, we've tried to keep the the API configuration minimal as this is an abstraction layer for location functionality and advanced users can always use other APIs to implement the details the way they want.
Also, we don't really have any cloud specific configurations in the location request configuration. We've been told that the use of nRF Cloud service is here mainly for testing and prototyping purposes and not that much for final products, which is another reason adding cloud service specific stuff would be nice to avoid.

Reply:
Excluding these configuration options from the location library does not work for us, since we changed our multi service sample to use the location library, as did others for ATv2. You are basically saying we have feature-frozen the positioning services as far as our samples are concerned.

Excluding these options also means you will not be able to exercise them from mosh. I would have thought you would be curious and interested to see how they work and want to try them.

We do still have the nrf_cloud_rest_cell_location sample which uses lower level positioning directly, but that one only does cell location not Wi-Fi (per the name).

Reply:
Replying quickly so you can still respond today...

> Excluding these configuration options from the location library does not work for us, since we changed our multi service sample to use the location library, as did others for ATv2. You are basically saying we have feature-frozen the positioning services as far as our samples are concerned.

I know this is a little complicated and would be nice to have all kinds of parameters in Location library (and we've discussed about many possible params related to cloud and other functionality such as NCELLMEAS search type) but:
- ATv2 is the only application in our SDK that uses Location library (I made the integration) and that's not a problem because it uses `CONFIG_LOCATION_SERVICE_EXTERNAL` and can hence send nRF Cloud request in the format it wants to.
- Having cloud service support in Location library has been questioned. There hasn't been serious discussions about removing it and I've always wanted to have it for quick testing and prototyping purposes. If the user wants to do something more, `CONFIG_LOCATION_SERVICE_EXTERNAL` is there.
- Multiservice sample could use `CONFIG_LOCATION_SERVICE_EXTERNAL` too instead of `CONFIG_LOCATION_SERVICE_NRF_CLOUD` if there are some cloud related functionalities not supported in Location library. But let's close this discussion first before rushing into it.
- If we would add these new cloud specific detailed parameters, I'd say there are probably dozen or two other parameters that are equally nice or important to have in the API. So this is why we try to carefully select the parameters we add to the API. And there are some that could probably be removed based on how things have evolved since the first versions of the API.

> Excluding these options also means you will not be able to exercise them from mosh. I would have thought you would be curious and interested to see how they work and want to try them.

We'll discuss this within our team but:
- In the past year or so, we've mainly sent direct REST requests to cloud through scripts using pre-collected NCELLMEAS results as an input. This way we can exercise more data points than using MOSH on a live device.
- We can still use any nRF Cloud Library API parameters through the `CONFIG_LOCATION_SERVICE_EXTERNAL` configuration, for which you also modified some default values.

Reply:
I've removed changes to the location library.